node-notes-cli/app.js

```diff
@@ -0,0 +1,58 @@
+import { readFile, writeFile } from 'node:fs/promises';
+
+const data = await JSON.parse(await readFile('data.json'));
+const [, , operation, ...args] = process.argv;
+
+function runApp(operation, args) {
+  switch (operation) {
+    case 'read':
+      readNotes(data);
+      break;
+    case 'create':
+      createNote(data, args);
+      break;
+    case 'update':
+      updateNote(data, args);
+      break;
+    case 'delete':
+      deleteNote(data, args);
+      break;
+  }
+}
+
+async function readNotes(data) {
+  for (const key in data.notes) {
+    console.log(key + ':', data.notes[key]);
+  }
+}
+
+async function createNote(data, args) {
+  const note = args[0];
+  const noteId = data.nextId;
+  data.notes[noteId] = note;
+  data.nextId++;
+  const dataJSON = JSON.stringify(data, null, 2);
+  await writeFile('data.json', dataJSON);
+}
+
+async function updateNote(data, args) {
+  for (const key in data.notes) {
+    if (key === args[0]) {
+      data.notes[key] = args[1];
+    }
+  }
+  const dataJSON = JSON.stringify(data, null, 2);
+  await writeFile('data.json', dataJSON);
+}
+
+async function deleteNote(data, args) {
+  delete data.notes[args[0]];
+  const dataJSON = JSON.stringify(data, null, 2);
+  await writeFile('data.json', dataJSON);
+}
+
+try {
+  runApp(operation, args);
+} catch (err) {
```

Comment:
This will never actually catch any exceptions because you aren't `await`ing a Promise. In asynchronous code, the only way to turn a Promise rejection into an exception is to `await` it.

However, you can't `await runApp(operation, args);` because `runApp` is not async and it returns `undefined` (since it doesn't have any return statements).

So, you need to make `runApp` async and `await` each of the calls to the async functions in the switch statement. Then you'll properly handle errors.